livecheck/main.py

```diff
@@ -471,22 +470,23 @@ def main(
     if auto_update and not os.access(search_dir, os.W_OK):
         raise click.ClickException(
             f'The directory "{working_dir}" must be writable because --auto-update is enabled.')
-    if git and not auto_update:
-        logger.error('Git option requires --auto-update')
-        raise click.Abort
     repo_root, repo_name = get_repository_root_if_inside(search_dir)
     if not repo_root:
         logger.error('Not inside a repository configured in repos.conf')
         raise click.Abort
-    if not os.path.isdir(os.path.join(repo_root, '.git')):
-        logger.error(f'Directory {repo_root} is not a git repository')
     if git:
+        if not auto_update:
+            logger.error('Git option requires --auto-update')
+            raise click.Abort
+        if not os.path.isdir(os.path.join(repo_root, '.git')):
+            logger.error(f'Directory {repo_root} is not a git repository')
+            raise click.Abort
         # Check if git is installed
-        if sp.run(('git', '--version'), stdout=sp.PIPE).returncode != 0:
+        if sp.run(('git', '--version'), stdout=sp.PIPE, check=True).returncode != 0:
```

Comment:
`check=True` won't work here. It will raise a `CalledProcessError` when the `returncode` is not zero. Use `check=False` or use `try/except`.